### PR TITLE
Move to pion/ice@v2

### PR DESCRIPTION
**Pull request by Sean-Der**
Removed support for trickle ice

### CODE REVIEW

**Comment by Sean-Der** [comment]
Everyone I added to the PR don't feel a need to review! I wanted to give everyone a heads up that Trickle ICE is now the default. This is a MASSIVE breaking change, so would love peoples input. 

This had to happen to make `ICE Restarts` possible. We need the ICEConnectionFailed state so people can handle things the same way they do in the browser.

After this lands I am going to update the README and start a wiki page about `v3 changes`. There is going to be a lot more happening https://github.com/pion/webrtc/issues/9

---

@at-wat w/e you think we are ready to tag `pion/ice@v2` then this is ready to merge.

**Comment by josharian** [comment]
May I suggest starting an “upgrading” doc explaining what people need to do to upgrade? I am happy to help copyedit it if you can get the rough material jotted down.

File: gathering_complete_promise.go [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+//
+// It is better to not use this function, and instead trickle candidates. If you use this function you will see longer connection startup times.
+// When the call is connected you will see no impact however.
+func GatheringCompletePromise(pc *PeerConnection) (gatherComplete <-chan struct{}) {
+	gatheringComplete, done := context.WithCancel(context.Background())
+
+	if pc.ICEGatheringState() == ICEGatheringStateComplete {
+		done()
+	} else {
+		pc.OnICECandidate(func(i *ICECandidate) {
```
[the remark this answers is not in the document]
Comment (Sean-Der) [reply gathering_complete_promise.go]: Done!

**Comment by Sean-Der** [comment]
@josharian done! https://github.com/pion/webrtc/wiki/Release-WebRTC@v3.0.0

**Comment by Sean-Der** [comment]
@at-wat I tagged `rc-1` for ICE! When this is approved I will push updates to the README.

Will put `ICE Restarts` in another PR. This is big enough already, but had to be done :)

File: go.mod [hunk 2]
```diff
@@ -16,4 +16,6 @@ require (
 	github.com/pion/transport v0.10.0
 	github.com/sclevine/agouti v3.0.0+incompatible
 	github.com/stretchr/testify v1.6.1
+	golang.org/x/net v0.0.0-20200625001655-4c5254603344 // indirect
+	golang.org/x/sys v0.0.0-20200625212154-ddb9806d33ae // indirect
```
Comment (at-wat) [inline go.mod:20]: seems be able to be removed.

**Pull request merged**